frontend/src/Login.tsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import './index.css';
+
+interface LoginProps {
+  login: (user: string, pass: string) => Promise<boolean>;
+  navigate: (r: string) => void;
+}
+
+interface LoginState {
+  username: string;
+  password: string;
+  errorMessage: string;
+}
+
+class Login extends React.Component<LoginProps, LoginState> {
+  constructor(props: LoginProps) {
+    super(props);
+    this.state = {
+      username: '',
+      password: '',
+      errorMessage: '',
+    };
+
+    this.setUsername = this.setUsername.bind(this);
+    this.setPassword = this.setPassword.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+    this.login = this.login.bind(this);
+  }
+
+  public onSubmit(event: any): void {
+    event.preventDefault();
+    this.login();
+  }
+
+  public setUsername(event: any): void {
+    this.setState({ username: event.target.value });
+  }
+
+  public setPassword(event: any): void {
+    this.setState({ password: event.target.value });
+  }
+
+  public async login() {
+    const { username, password } = this.state;
+    const { login, navigate } = this.props;
+    const success = await login(username, password);
+    if (success) {
+      navigate('/dashboard');
+    } else {
+      this.setState({ errorMessage: 'Invalid email/password.' });
+    }
```

Comment:
lets add a proper error handling, right now all API errors get swallowed by the `api.tsx` so I suggest to propagate an actual error and display it to the user.